File: Tests/fableBuild/Tests.Antidote.uFuzzy.js
import ufuzzy from "@leeoniya/ufuzzy";
import { map, toArray } from "./fable_modules/fable-library.3.7.8/Seq.js";
import { assertEqual } from "./fable_modules/fable-library.3.7.8/Util.js";

export const haystack = ["puzzle", "Super Awesome Thing (now with stuff!)", "FileName.js", "/feeding/the/catPic.jpg"];

export const needle = "feed cat";

describe("uFuzzy.search", () => {
    it("works with strings", () => {
        const uf = new ufuzzy({});
        const result = toArray(map((x) => haystack[~(~x)], uf.filter(haystack, needle)));
        assertEqual(result[0], "/feeding/the/catPic.jpg");
    });
});

File: Tests/Tests.Antidote.uFuzzy.fs
module Tests.Fuse

open Mocha
open Node
open Fable.Core
open Fable.Core.Testing
open Fable.Core.JsInterop
open Antidote.UFuzzy


let haystack = [|
    "puzzle"
    "Super Awesome Thing (now with stuff!)"
    "FileName.js"
    "/feeding/the/catPic.jpg"
|]

let needle = "feed cat"

describe """uFuzzy.search""" (fun () ->

    it "works with strings" (fun () ->

        let uf = uFuzzy.Create( Antidote.UFuzzy.UFuzzy.IUFuzzyOptions() )

        let idxs = uf.filter(haystack, needle)
        let result = idxs |> Seq.map(fun x -> haystack.[int x]) |> Seq.toArray
        Assert.AreEqual(result.[0] , "/feeding/the/catPic.jpg")
    )
)
